### ISSUE TITLE
SDK 49 regression -- Android expo-dev-client crashes on builds without hermes

### ISSUE THREAD

**Issue by evelant** [expo/expo#23312]
### Summary

Using SDK 49 beta 5 expo-dev-client crashes on launch with `couldn't find libhermes.so` in builds that don't use hermes.

Reproduction: https://github.com/evelant/expo-dev-client-no-hermes-bug

![Screenshot_20230705-151320_expo-dev-client-no-hermes-bug](https://github.com/expo/expo/assets/858965/c36bed6f-16f0-4afb-ade8-f6523a713679)

### Managed or bare workflow?

managed

### What platform(s) does this occur on?

Android

### Package versions

```
  "dependencies": {
    "expo": "~49.0.0-beta.5",
    "expo-status-bar": "~1.6.0",
    "react": "18.2.0",
    "react-native": "0.72.1",
    "react-native-v8": "2.2.0",
    "v8-android-jit": "11.1000.3",
    "expo-dev-client": "~2.4.3"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "expo-dev-launcher": "2.4.5",
    "expo-dev-menu": "3.1.5",
    "expo-dev-menu-interface": "1.3.0",
    "@expo/config-plugins": "~7.2.5"
  },
```

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.4.1
      Shell: 0.80.0 - /Users/imagio/.cargo/bin/nu
    Binaries:
      Node: 18.16.0 - ~/Library/Caches/fnm_multishells/64229_1688522408590/bin/node
      Yarn: 1.22.19 - ~/Library/pnpm/yarn
      npm: 9.7.2 - ~/Library/pnpm/npm
    Managers:
      CocoaPods: 1.11.3 - /Users/imagio/.asdf/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Android Studio: Flamingo 2022.2.1 Patch 2 Flamingo 2022.2.1 Patch 2
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      expo: ~49.0.0-beta.5 => 49.0.0
      react: 18.2.0 => 18.2.0
      react-native: 0.72.1 => 0.72.1
    npmGlobalPackages:
      expo-cli: 6.3.9
    Expo Workflow: managed
```

### Reproducible demo

https://github.com/evelant/expo-dev-client-no-hermes-bug

### Stacktrace (if a crash is involved)

See screenshot above

**Comment by evelant** [comment]
Likely related https://github.com/expo/expo/issues/17450 -- although the fix suggested there of adding `android.packagingOptions.jniLibs.useLegacyPackaging = true` does not solve the issue in this case.

**Comment by evelant** [comment]
@Kudo This also appears to happen when using JSC. Any build with dev-client will crash because it can't load hermes.

**Comment by Kudo** [comment]
@evelant good catch, much appreciated for that. that issue will be resolved by #23415
